### PR TITLE
PYIC-4811: Add smartphone triage page

### PR DESCRIPTION
**Pull request by MikeCollingwood**
### What changed

- Add smartphone triage page

### Why did it change

- Migrating pages from mobile

### Issue tracking

- [PYIC-4811](https://govukverify.atlassian.net/browse/PYIC-4811)

Screenshot 2024-03-20 at 10 17 30

[PYIC-4811]: https://govukverify.atlassian.net/browse/PYIC-4811?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/views/ipv/page/pyi-triage-select-smartphone.njk [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+      ]
+    } %}
+
+    {% if errorState %}
+      {% set errorMessageObject = { 'text': 'pages.pyiTriageSelectSmartphone.content.formErrorMessage.errorRadioMessage' | translate } %}
+      {% set radiosConfig = radiosConfig | setAttribute('errorMessage', errorMessageObject) %}
+    {% endif %}
+
+    {{ govukRadios(radiosConfig) }}
+    <button name="submitButton" class="govuk-button" data-module="govuk-button" id="submitButton">
```
Comment (MikeCollingwood) [inline src/views/ipv/page/pyi-triage-select-smartphone.njk:79]: use govukButton

File: src/views/ipv/page/pyi-triage-select-smartphone.njk [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+{% set errorText = 'pages.pyiTriageSelectSmartphone.content.formErrorMessage.errorSummaryDescriptionText' | translate %}
+{% set errorHref = "#pyiTriageSelectSmartphoneOptionsForm" %}
+
+{% block content %}
+  <h1 class="govuk-heading-l" id="header" data-page="{{ googleTagManagerPageId }}">{{ 'pages.pyiTriageSelectSmartphone.header' | translate }}</h1>
+  <p class="govuk-body">{{ 'pages.pyiTriageSelectSmartphone.content.paragraphIntro1' | translate }}</p>
+  <p class="govuk-body">{{ 'pages.pyiTriageSelectSmartphone.content.paragraphIntro2' | translate }}</p>
+
+  <h2 class="govuk-heading-m govuk-!-margin-top-7">{{ 'pages.pyiTriageSelectSmartphone.content.iphone.subHeading' | translate }}</h2>
+  <p class="govuk-body govuk-!-margin-bottom-2">{{ 'pages.pyiTriageSelectSmartphone.content.iphone.paragraph' | translate }}</p>
```
Comment (AmritSidhu) [inline src/views/ipv/page/pyi-triage-select-smartphone.njk]: Is this necessary `govuk-!-margin-bottom-2`? I think its causing a big gap in-between the text when displayed on the page. Looks like another occurrence on the page to

File: src/views/ipv/page/pyi-triage-select-smartphone.njk [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+      ]
+    } %}
+
+    {% if errorState %}
+      {% set errorMessageObject = { 'text': 'pages.pyiTriageSelectSmartphone.content.formErrorMessage.errorRadioMessage' | translate } %}
+      {% set radiosConfig = radiosConfig | setAttribute('errorMessage', errorMessageObject) %}
+    {% endif %}
+
+    {{ govukRadios(radiosConfig) }}
+    {{ govukButton({
```
Comment (AmritSidhu) [inline src/views/ipv/page/pyi-triage-select-smartphone.njk:81]: I dont think this is the button we want here. We use this on pages where it is unrecoverable and direct the user to gov.uk

File: src/views/ipv/page/pyi-triage-select-smartphone.njk (hunk 2 above)
Comment (MikeCollingwood) [reply src/views/ipv/page/pyi-triage-select-smartphone.njk]: Which gap do you mean? I think `govuk-!-margin-top-7` might be to blame for the biggest gap above the iPhone subheading. I've removed them now:
Screenshot 2024-03-19 at 11 37 38

File: src/views/ipv/page/pyi-triage-select-smartphone.njk (hunk 3 above)
Comment (MikeCollingwood) [reply src/views/ipv/page/pyi-triage-select-smartphone.njk:81]: Issue references [here](https://github.com/govuk-one-login/ipv-core-front/pull/1339#discussion_r1530165985)

Comment (blakeyp) [reply src/views/ipv/page/pyi-triage-select-smartphone.njk:81]: I think you'll need to add `id: submitButton` for the event listener which disables the button on click

**Review by AmritSidhu — DISMISSED** [review]

File: src/views/ipv/page/pyi-triage-select-smartphone.njk [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+      {% set errorMessageObject = { 'text': 'pages.pyiTriageSelectSmartphone.content.formErrorMessage.errorRadioMessage' | translate } %}
+      {% set radiosConfig = radiosConfig | setAttribute('errorMessage', errorMessageObject) %}
+    {% endif %}
+
+    <hr class="govuk-section-break govuk-section-break--l govuk-section-break--visible">
+
+    {{ govukRadios(radiosConfig) }}
+    {{ govukButton({
+      id: submitButton,
+      classes: "govuk-!-margin-top-7",
```
Comment (DanCorderIPV) [inline src/views/ipv/page/pyi-triage-select-smartphone.njk]: I suspect that the spacing above the button in the desktop UCD screenshot is not deliberate - there is no gap above the button in the mobile version of the page (and the gap looks weird to me anyway).

**Review by DanCorderIPV — CHANGES_REQUESTED** [review]

File: src/views/ipv/page/pyi-triage-select-smartphone.njk (hunk 4 above)
Comment (AmritSidhu) [reply src/views/ipv/page/pyi-triage-select-smartphone.njk]: Yeah agreed, this class should be removed

Comment (MikeCollingwood) [reply src/views/ipv/page/pyi-triage-select-smartphone.njk]: > I suspect that the spacing above the button in the desktop UCD screenshot is not deliberate - there is no gap above the button in the mobile version of the page (and the gap looks weird to me anyway).

Glad you think so - will remove

**Comment by MikeCollingwood** [comment]
Screenshot 2024-03-20 at 10 40 27

**Review by DanCorderIPV — APPROVED** [review]

**Pull request merged**